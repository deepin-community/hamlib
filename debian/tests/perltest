#!/bin/sh

set -eu

trap "rm -f perltest.out" EXIT

perl /usr/share/doc/libhamlib-perl/examples/perltest.pl > perltest.out
cat perltest.out

# paper over precision difference on i386
#-Latitude:	  48.1042,   48Â°  6' 15" N	recoded:   48.1042
#+Latitude:	  48.1042,   48Â°  6' 14" N	recoded:   48.1039
sed -i -e 's/ 14" N/ 15" N/' -e 's/48.1039/48.1042/' \
	-e 's/^\(Backend version:	\).*/\1xxx.x/' \
	perltest.out

diff -u debian/tests/perltest.out perltest.out
