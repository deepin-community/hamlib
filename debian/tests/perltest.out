get_conf:		path = "/dev/Rig", retry = 50
freq:			14266000
get_mode:		FM
width:			15000
get_vfo:		Main
Backend copyright:	LGPL
Model:			Dummy
Manufacturer:		Hamlib
Backend version:	xxx.x
get_info:		Nothing much (dummy)
VOX delay:		0
VOX delay:		5
get_channel status:	0 = rig_ext_lookup called
VFO:			VFOA, 14266000
Attenuators:		10 20 30 0 0 0 0 0

Sending Morse, '73'

Some static functions:
Loc1:		IN98XC ->   -0.0417,   48.1042-> IN98XC
Loc1:		DM33DX -> -113.7083,   33.9792-> DM33DX
Distance:	8765.814 km, azimuth 309.00, long path: 31266.186 km
Longitude:	  -0.0417,    0°  2' 29" W	recoded:   -0.0414
Latitude:	  48.1042,   48°  6' 15" N	recoded:   48.1042
